### PR TITLE
add UI registration and confirm_email

### PR DESCRIPTION
**Pull request by ykobuk**
Added ui_registration and ui_confirm_email.

### CODE REVIEW

File: client/src/components/Nav/ConfirmEmail.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React, { Component } from 'react';
```
Comment (Merteg) [inline client/src/components/Nav/ConfirmEmail.jsx]: I think you don't need to store this file in Nav directory

File: client/src/components/Nav/ConfirmEmail.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+                this.setState({msg: data.msg})
+                window.Materialize.toast(this.state.msg, 3000)
+            })
+    }
+    render() {
+        return <p></p>
+    }
+}
+
+export default ConfirmEmail;
```
Comment (Merteg) [inline client/src/components/Nav/ConfirmEmail.jsx]: No new line at end of the file

File: client/src/components/Nav/Registration.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react';
+import { Input, Button, Icon, Row } from 'react-materialize'
```
Comment (Merteg) [inline client/src/components/Nav/Registration.jsx]: You don't use Icon component here

File: client/src/components/Nav/Registration.jsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            label="password"
+            m={6}/>
+            <Button waves="light" onClick={this.register}>SignUp</Button>
+        </Row>
+
+        );}
+    }
+
+
+export default Registration;
```
Comment (Merteg) [inline client/src/components/Nav/Registration.jsx]: Also, no new line

File: client/src/components/Nav/index.jsx [hunk 5]
```diff
@@ -1,9 +1,10 @@
 import React, { Component } from 'react';
-import { Navbar } from 'react-materialize';
+import { Navbar, Button } from 'react-materialize';
 import LogIn from './LogIn';
 import NavProfile from './NavProfile';
 import LogOut from './LogOut';
 import {isLogged} from '../../utils'
+import {NavLink} from 'react-router-dom'
```
Comment (Merteg) [inline client/src/components/Nav/index.jsx:7]: In my opinion better to group and sort imports like we did it in python.

File: client/src/components/Nav/index.jsx [hunk 6]
```diff
@@ -25,7 +26,8 @@ class Nav extends Component {
         </React.Fragment>
       )
     }
-    else return <LogIn update={this.update} />
+    else return (<React.Fragment><LogIn update={this.update} /><Button
```
Comment (Merteg) [inline client/src/components/Nav/index.jsx]: Please, don't write all components at one line, it's really hard to read.

File: server/server/views/registration.py [hunk 7]
```diff
@@ -9,15 +9,30 @@
...
+                   factory=RegistrateUser,
+                   cors_origins=('http://localhost:3000',))
 confirm_register = Service(name='email_confirm',
-                           path='/email_confirm/{email_confirm}')
+                           path='/email_confirm/{email_confirm}',
+                           factory=RegistrateUser,
+                           cors_origins=('http://localhost:3000',))
 
 
 @register.post()
```
Comment (Merteg) [inline server/server/views/registration.py:39]: I think, this view seems better at user resource, in accordance to REST, post - Create.
Also I can refer to Sergiy comments at PR #15

File: server/server/views/registration.py [hunk 8]
```diff
@@ -66,7 +79,7 @@ def confirm_registration_view(request):
     user_email_token = request.matchdict['email_confirm']
     non_active_user = User.get_one(request, url_token=user_email_token)
     if non_active_user is None:
-        return HTTPNotFound()
+        return {"msg": "Error404 HTTPNotFound"}
```
Comment (Merteg) [inline server/server/views/registration.py:83]: Why did you remove that?

File: client/src/components/Nav/Registration.jsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+            type="email"
+            label="email"
+            m={6}/>
+          <Input
+            id="password"
+            value={this.state.password}
+            onChange={this.onChangeHandler}
+            placeholder="Password"
+            type="password"
+            label="password"
```
Comment (Merteg) [inline client/src/components/Nav/Registration.jsx]: I think Password in label seems better

**Review by Merteg — CHANGES_REQUESTED** [review]
Also, you don't have validation at server

File: server/server/views/registration.py (hunk 8 above)
Comment (ykobuk) [reply server/server/views/registration.py:83]: Cause front doesn't return this Error yet, and this way simple to show this Error, and everyone will know what it does mean

File: client/src/components/Nav/Registration.jsx [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+
+    render() {
+      return (
+        <Row>
+          <h3>Registration form</h3>
+          <Input
+            id="email"
+            velue={this.state.email}
```
Comment (RayMet) [inline client/src/components/Nav/Registration.jsx]: You have wrong name of value

**Review by RayMet — CHANGES_REQUESTED** [review]

File: client/src/components/Nav/Registration.jsx (hunk 10 above)
Comment (ykobuk) [reply client/src/components/Nav/Registration.jsx]: Why do you think that?

Comment (RayMet) [reply client/src/components/Nav/Registration.jsx]: "velue" correct? or value?

**Comment by ykobuk** [comment]
Add the repeat_password field on the back and front, please check it please.

**Review by RayMet — APPROVED** [review]

File: client/src/components/ConfirmEmail.jsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+import {request} from '../utils';
+import {Home} from "./Home";
+
+class ConfirmEmail extends Component{
+    state = {
+        msg: ''
+    }
+
+    componentDidMount(){
+        request('/email_confirm/'+this.props.match.params.token, "GET")
```
Comment (Merteg) [inline client/src/components/ConfirmEmail.jsx]: GET is a value by default at this function, you don't need to type this again.

File: client/src/components/ConfirmEmail.jsx [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+            .then(data=>{
+                this.setState({msg: data.msg})
+                window.Materialize.toast(this.state.msg, 3000)
+            })
+    }
+    render() {
+        return <p></p>
```
Comment (Merteg) [inline client/src/components/ConfirmEmail.jsx]: It is don't very important, but you can render React.Fragment against of `<p>`, or maybe `<div>`. It is my opinion, not very important)

File: client/src/components/ConfirmEmail.jsx [hunk 13]
```diff
@@ -0,0 +1,22 @@
+import React, { Component } from 'react';
+import {request} from '../utils';
```
Comment (Merteg) [inline client/src/components/ConfirmEmail.jsx]: In accordance to style code [Add spaces inside curly braces](https://github.com/airbnb/javascript/#whitespace--in-braces)

File: client/src/components/Nav/Registration.jsx [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+        this.setState({ [id]: e.currentTarget.value })
+    };
+
+    register = (e) => {
+        let register_data = {
+            email: this.state.email,
+            password: this.state.password,
+            repeat_password: this.state.repeat_password
+        }
+        if (!emailValidation(this.state.email)
```
Comment (Merteg) [inline client/src/components/Nav/Registration.jsx]: [Style code](https://github.com/airbnb/javascript/#control-statements) The logical operator should begin the line.
P. S. I must to change this too ))

File: client/src/components/Nav/Registration.jsx [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+            value={this.state.repeat_password}
+            onChange={this.onChangeHandler}
+            placeholder="Repeat password"
+            type="password"
+            label="Repeat password"
+            m={12}/>
+            <Button waves="light" onClick={this.register}>SignUp</Button>
+        </Row>
+
+        );}
```
Comment (Merteg) [inline client/src/components/Nav/Registration.jsx]: This hurt my eyes a little bit. Can you make it like this?
```
    );
}
```

**Review by Merteg — CHANGES_REQUESTED** [review]
Will you change service to resource?)

File: client/src/components/Nav/Registration.jsx (hunk 14 above)
Comment (Merteg) [reply client/src/components/Nav/Registration.jsx]: I was wrong, it's all okey here. One thing you can do better, it's move `!emailValidation(this.state.email)` to the new line like at style code guide

**Comment by ykobuk** [comment]
Fix ConfirmEmail.jsx and Registration.jsx"

**Review by Merteg — APPROVED** [review]
I approve this.
